hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/services/DefinitionsService.java

```diff
@@ -37,31 +37,132 @@
-    Collection<Trigger> getTriggers() throws Exception;
+
+    /**
+     * Update the <code>Trigger</code>. <code>Conditions</code> and <code>Actions</code> are manipulated in separate
+     * calls. The updated </code>Trigger</code> will be persisted.  If enabled the </code>Trigger</code>
+     * will be [re-]inserted into the Alerts engine and any prior dampening will be reset.
+     * @param triggerId
+     * @throws Exception If the </code>Trigger</code> does not exist.
+     */
+    Trigger updateTrigger(Trigger trigger) throws Exception;
+
     Trigger getTrigger(String triggerId) throws Exception;
 
+    Collection<Trigger> getAllTriggers() throws Exception;
+
     /*
-        CRUD interface for Trigger
+        CRUD interface for Dampening
+     */
+
+    Dampening addDampening(Dampening dampening) throws Exception;
+
+    void removeDampening(String dampeningId) throws Exception;
+
+    Dampening updateDampening(Dampening dampening) throws Exception;
+
+    Dampening getDampening(String dampeningId) throws Exception;
+
+    /**
+     * @param triggerId
+     * @param triggerMode Return only dampenings for the given trigger mode. Return all if null.
+     * @return The existing dampenings for the trigger. Not null.
+     * @throws Exception
      */
-    void addDampening(Dampening dampening) throws Exception;
-    void removeDampening(String triggerId) throws Exception;
-    void updateDampening(Dampening dampening) throws Exception;
-    Collection<Dampening> getDampenings() throws Exception;
-    Dampening getDampening(String triggerId) throws Exception;
+    Collection<Dampening> getTriggerDampenings(String triggerId, Trigger.Mode triggerMode) throws Exception;
+
+    Collection<Dampening> getAllDampenings() throws Exception;
 
     /*
         CRUD interface for Condition
      */
-    void addCondition(Condition condition) throws Exception;
-    void removeCondition(String conditionId) throws Exception;
-    void updateCondition(Condition condition) throws Exception;
-    Collection<Condition> getConditions() throws Exception;
-    Collection<Condition> getConditions(String triggerId) throws Exception;
+
+    /**
+     * A convenience method that adds a new Condition to the existing condition set for the specified
+     * Trigger and trigger mode.  The new condition will be assigned the highest conditionSetIndex for the
+     * updated conditionSet. The following Condition fields are ignored for the
+     * incoming condition, and set in the returned collection set:<pre>
+     *   conditionId
+     *   triggerId
+     *   triggerMode
+     *   conditionSetSize
+     *   conditionSetIndex
+     * </pre>
+     * @param triggerId
+     * @param triggerMode
+     * @param condition Not null
+     * @return The updated, persisted condition set
+     * @throws Exception
+     */
+    Collection<Condition> addCondition(String triggerId, Trigger.Mode triggerMode, Condition condition)
```

Comment:
+1
I think yesterday I was misunderstanding the chat.
So +1 to be able to add a condition with the triggerId and the new Condition object and it's a good idea to have the updated persisted condition set.